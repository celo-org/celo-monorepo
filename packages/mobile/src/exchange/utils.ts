import BigNumber from 'bignumber.js'
import { CURRENCY_ENUM } from 'src/geth/consts'
import { convertToContractDecimals } from 'src/tokens/saga'
import { getContractKitAsync } from 'src/web3/contracts'

export async function celoTocUsd(celoAmount: BigNumber) {
  const celoAmountInWei = (
    await convertToContractDecimals(celoAmount, CURRENCY_ENUM.GOLD)
  ).integerValue()

  const contractKit = await getContractKitAsync()
  const exchange = await contractKit.contracts.getExchange()
  const exchangeRate = await exchange.getGoldExchangeRate(celoAmountInWei)
  const cUsdAmount = exchangeRate.multipliedBy(celoAmount)

  return cUsdAmount
}
