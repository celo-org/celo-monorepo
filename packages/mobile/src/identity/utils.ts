import BigNumber from 'bignumber.js'

export function isUserBalanceSufficient(
  userBalance: string | number | null,
  estimatedTxAmount: number | string
) {
  if (!userBalance || new BigNumber(userBalance).isLessThan(new BigNumber(estimatedTxAmount))) {
    return false
  }

  return true
}
