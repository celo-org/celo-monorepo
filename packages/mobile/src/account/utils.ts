import { ADDRESS_LENGTH } from 'src/exchange/reducer'

export const isAddressFormat = (content: string): boolean => {
  return content.startsWith('0x') && content.length === ADDRESS_LENGTH
}
