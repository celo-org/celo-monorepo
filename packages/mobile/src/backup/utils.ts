import { sampleSize } from 'lodash'
import { AsyncStorage } from 'react-native'
import { generateMnemonic, wordlists } from 'react-native-bip39'
import { getKey, setKey } from 'src/utils/keyStore'
import Logger from 'src/utils/Logger'

const TAG = 'Backup/utils'

export async function createQuizWordList(mnemonic: string, language: string | null) {
  const disallowedWordSet = new Set(mnemonic.split(' '))
  const languageWordList = getWordlist(language)
  const wordOptions: string = await generateMnemonic(1000, null, languageWordList)
  const quizWordList = new Set(
    [...wordOptions.split(' ')].filter((word: string) => !disallowedWordSet.has(word))
  )
  return [...quizWordList]
}

export function selectQuizWordOptions(correctWord: string, allWords: string[], numOptions: number) {
  const wordOptions = []
  const correctWordPosition = Math.floor(Math.random() * numOptions)
  const randomWordIndexList = sampleSize([...Array(allWords.length).keys()], numOptions - 1)
  let randomWordIndex: number = 0

  for (let i = 0; i < numOptions; i++) {
    if (i === correctWordPosition) {
      wordOptions.push(correctWord)
      continue
    }

    wordOptions.push(allWords[randomWordIndexList[randomWordIndex]])
    randomWordIndex += 1
  }
  return wordOptions
}

export function getWordlist(language: string | null) {
  if (!language) {
    return wordlists.EN
  }

  switch (language.slice(0, 2)) {
    case 'es': {
      return wordlists.ES
    }
    default: {
      return wordlists.EN
    }
  }
}

// TODO(Rossy) Remove after the next alfa testnet reset
export async function getStoredMnemonic(): Promise<string | null> {
  try {
    Logger.debug(TAG, 'Checking keystore for mnemonic')
    let mnemonic = await getKey('mnemonic')
    if (mnemonic) {
      return mnemonic
    }

    Logger.debug(TAG, 'Mnemonic not found in keystore, checking async storage')
    mnemonic = await AsyncStorage.getItem('mnemonic')
    if (mnemonic) {
      await setKey('mnemonic', mnemonic)
      await AsyncStorage.removeItem('mnemonic')
      return mnemonic
    }

    Logger.error(TAG, 'No mnemonic found')
    return null
  } catch (error) {
    Logger.error(TAG, 'Failed to retrieve mnemonic', error)
    return null
  }
}
