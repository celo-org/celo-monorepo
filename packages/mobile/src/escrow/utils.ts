import { ensureLeading0x } from '@celo/base'
import { generateDeterministicInviteCode } from '@celo/utils/lib/account'
import { publicKeyToAddress } from '@celo/utils/lib/address'

export const generateEscrowPaymentIdAndPk = (
  recipientPhoneHash: string,
  recipientPepper: string,
  addressIndex: number = 0
) => {
  const { publicKey, privateKey } = generateDeterministicInviteCode(
    recipientPhoneHash,
    recipientPepper,
    addressIndex
  )
  return { paymentId: publicKeyToAddress(publicKey), privateKey: ensureLeading0x(privateKey) }
}

export const generateUniquePaymentId = (
  existingPaymentIds: string[],
  phoneHash: string,
  pepper: string
) => {
  const paymentIdSet: Set<string> = new Set(existingPaymentIds)

  // Using an upper bound of 100 to be sure this doesn't run forever given
  // the realistic amount of pending escrow txs is far less than this
  for (let i = 0; i < 100; i += 1) {
    const { paymentId } = generateEscrowPaymentIdAndPk(phoneHash, pepper, i)
    if (!paymentIdSet.has(paymentId)) {
      return paymentId
    }
  }
}
