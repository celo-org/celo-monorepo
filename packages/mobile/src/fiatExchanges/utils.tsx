import { SIMPLEX_URI } from 'src/config'
import { CURRENCY_ENUM } from 'src/geth/consts'
import { LocalCurrencyCode } from 'src/localCurrency/consts'
import { navigate } from 'src/navigator/NavigationService'
import { Screens } from 'src/navigator/Screens'
import { navigateToURI } from 'src/utils/linking'

export const openMoonpay = (currencyCode: LocalCurrencyCode, currencyToBuy: CURRENCY_ENUM) => {
  navigate(Screens.MoonPay, {
    localAmount: 0,
    currencyCode,
    currencyToBuy,
  })
}

export const openSimplex = (account: string | null) => {
  navigateToURI(`${SIMPLEX_URI}?address=${account}`)
}
