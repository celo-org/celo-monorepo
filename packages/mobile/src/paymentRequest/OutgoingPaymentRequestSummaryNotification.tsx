import * as React from 'react'
import { WithNamespaces, withNamespaces } from 'react-i18next'
import { Image, StyleSheet } from 'react-native'
import { connect } from 'react-redux'
import { PaymentRequest } from 'src/account/types'
import CeloAnalytics from 'src/analytics/CeloAnalytics'
import { CustomEventNames } from 'src/analytics/constants'
import { updatePaymentRequestNotified, updatePaymentRequestStatus } from 'src/firebase/actions'
import { Namespaces } from 'src/i18n'
import {
  addressToE164NumberSelector,
  AddressToE164NumberType,
  e164NumberToAddressSelector,
  E164NumberToAddressType,
} from 'src/identity/reducer'
import { sendDollar } from 'src/images/Images'
import { navigate } from 'src/navigator/NavigationService'
import { Stacks } from 'src/navigator/Screens'
import SummaryNotification from 'src/notifications/SummaryNotification'
import { listItemRenderer } from 'src/paymentRequest/OutgoingPaymentRequestListScreen'
import PaymentRequestNotificationInner from 'src/paymentRequest/PaymentRequestNotificationInner'
import { NumberToRecipient, phoneNumberToRecipient } from 'src/recipients/recipient'
import { recipientCacheSelector } from 'src/recipients/reducer'
import { RootState } from 'src/redux/reducers'

interface OwnProps {
  requests: PaymentRequest[]
}

interface DispatchProps {
  updatePaymentRequestStatus: typeof updatePaymentRequestStatus
  updatePaymentRequestNotified: typeof updatePaymentRequestNotified
}

type Props = OwnProps & DispatchProps & WithNamespaces & StateProps

interface StateProps {
  e164PhoneNumberAddressMapping: E164NumberToAddressType
  addressToE164Number: AddressToE164NumberType
  recipientCache: NumberToRecipient
}

const mapStateToProps = (state: RootState): StateProps => ({
  e164PhoneNumberAddressMapping: e164NumberToAddressSelector(state),
  addressToE164Number: addressToE164NumberSelector(state),
  recipientCache: recipientCacheSelector(state),
})

// Payment Request notification for the notification center on home screen
export class OutgoingPaymentRequestSummaryNotification extends React.Component<Props> {
  onReview = () => {
    CeloAnalytics.track(CustomEventNames.outgoing_request_payment_review)
    navigate(Stacks.OutgoingRequestStack)
  }

  getRequesterRecipient = (requesterE164Number: string) => {
    return phoneNumberToRecipient(
      requesterE164Number,
      this.props.e164PhoneNumberAddressMapping[requesterE164Number],
      this.props.recipientCache
    )
  }

  itemRenderer = (item: PaymentRequest) => {
    return (
      <PaymentRequestNotificationInner
        key={item.uid}
        amount={item.amount}
        requesterE164Number={item.requesterE164Number}
        requesterRecipient={this.getRequesterRecipient(item.requesterE164Number)}
      />
    )
  }

  render() {
    const { recipientCache, requests, t } = this.props
    return requests.length === 1 ? (
      listItemRenderer({
        addressToE164Number: this.props.addressToE164Number,
        updatePaymentRequestStatus: this.props.updatePaymentRequestStatus,
        updatePaymentRequestNotified: this.props.updatePaymentRequestNotified,
        recipientCache,
      })(requests[0])
    ) : (
      <SummaryNotification<PaymentRequest>
        items={requests}
        title={t('outgoingPaymentRequests')}
        icon={<Image source={sendDollar} style={styles.image} resizeMode="contain" />}
        onReview={this.onReview}
        itemRenderer={this.itemRenderer}
      />
    )
  }
}

const styles = StyleSheet.create({
  image: {
    width: 40,
    height: 40,
  },
})

export default connect<StateProps, DispatchProps, {}, RootState>(
  mapStateToProps,
  {
    updatePaymentRequestStatus,
    updatePaymentRequestNotified,
  }
)(withNamespaces(Namespaces.walletFlow5)(OutgoingPaymentRequestSummaryNotification))
