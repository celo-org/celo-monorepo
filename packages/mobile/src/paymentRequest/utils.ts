import { PaymentRequest } from 'src/account/types'
import { AddressToE164NumberType } from 'src/identity/reducer'
import { NumberToRecipient, Recipient, RecipientKind } from 'src/recipients/recipient'

export function getRecipientFromPaymentRequest(
  paymentRequest: PaymentRequest,
  recipientCache: NumberToRecipient
): Recipient {
  const cachedRecipient = paymentRequest.requesterE164Number
    ? recipientCache[paymentRequest.requesterE164Number]
    : null

  if (cachedRecipient) {
    return {
      ...cachedRecipient,
      kind: RecipientKind.Address,
      address: paymentRequest.requesterAddress,
    }
  } else {
    return {
      kind: RecipientKind.Address,
      address: paymentRequest.requesterAddress,
      displayName: paymentRequest.requesterE164Number || paymentRequest.requesterAddress,
    }
  }
}

export function getSenderFromPaymentRequest(
  paymentRequest: PaymentRequest,
  addressToE164Number: AddressToE164NumberType,
  recipientCache: NumberToRecipient
): Recipient {
  const e164PhoneNumber = addressToE164Number[paymentRequest.requesteeAddress]
  if (!e164PhoneNumber) {
    return {
      kind: RecipientKind.Address,
      address: paymentRequest.requesteeAddress,
      displayName: paymentRequest.requesteeAddress,
    }
  }

  const cachedRecipient = recipientCache[e164PhoneNumber]
  if (cachedRecipient) {
    return {
      ...cachedRecipient,
      kind: RecipientKind.Address,
      address: paymentRequest.requesteeAddress,
    }
  } else {
    return {
      kind: RecipientKind.MobileNumber,
      address: paymentRequest.requesterAddress,
      e164PhoneNumber,
      displayName: e164PhoneNumber,
    }
  }
}
