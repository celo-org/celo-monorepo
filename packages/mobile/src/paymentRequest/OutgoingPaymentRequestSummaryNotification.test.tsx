import * as React from 'react'
import 'react-native'
import { Provider } from 'react-redux'
import * as renderer from 'react-test-renderer'
import OutgoingPaymentRequestSummaryNotification from 'src/paymentRequest/OutgoingPaymentRequestSummaryNotification'
import { createMockStore } from 'test/utils'
import { mockE164Number, mockPaymentRequests } from 'test/values'

const store = createMockStore()

describe('OutgoingPaymentRequestSummaryNotification', () => {
  it('renders correctly for multiple requests', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <OutgoingPaymentRequestSummaryNotification requests={mockPaymentRequests} />
      </Provider>
    )
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly for just one alone', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <OutgoingPaymentRequestSummaryNotification requests={mockPaymentRequests.slice(0, 1)} />
      </Provider>
    )
    expect(tree).toMatchSnapshot()
  })

  it('renders a number when the address mapping is cached', () => {
    const storeWithMapping = createMockStore({
      identity: { addressToE164Number: { mockAccount: mockE164Number } },
    })
    const tree = renderer.create(
      <Provider store={storeWithMapping}>
        <OutgoingPaymentRequestSummaryNotification requests={mockPaymentRequests.slice(0, 1)} />
      </Provider>
    )
    expect(tree).toMatchSnapshot()
  })
})
