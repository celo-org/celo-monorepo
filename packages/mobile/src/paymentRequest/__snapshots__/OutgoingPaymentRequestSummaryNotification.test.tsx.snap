// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutgoingPaymentRequestSummaryNotification renders a number when the address mapping is cached 1`] = `
<View>
  <View
    style={
      Object {
        "marginBottom": 16,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "padding": 16,
          },
          Object {
            "borderRadius": 8,
          },
          Object {
            "elevation": 12,
            "shadowColor": "rgba(156, 164, 169, 0.4)",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 12,
          },
          Array [
            Object {
              "height": 144,
            },
            Object {},
          ],
        ]
      }
      testID="OutgoingPaymentRequestNotification/FAKE_ID_1"
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "marginRight": 16,
            }
          }
        >
          <Text
            numberOfLines={2}
            style={
              Object {
                "color": "#9CA4A9",
                "fontFamily": "Inter-Regular",
                "fontSize": 13,
                "lineHeight": 16,
                "marginBottom": 4,
              }
            }
            testID="OutgoingPaymentRequestNotification/FAKE_ID_1/Title"
          >
            outgoingPaymentRequestNotificationTitle
          </Text>
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "#2E3338",
                "fontFamily": "Inter-SemiBold",
                "fontSize": 24,
                "lineHeight": 28,
                "marginBottom": 2,
              }
            }
            testID="OutgoingPaymentRequestNotification/FAKE_ID_1/Amount"
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  undefined,
                  Object {
                    "color": undefined,
                  },
                ]
              }
            >
              
              $
              266,000.00
            </Text>
          </Text>
          <Text
            style={
              Object {
                "color": "#2E3338",
                "fontFamily": "Inter-Regular",
                "fontSize": 14,
                "lineHeight": 18,
              }
            }
            testID="OutgoingPaymentRequestNotification/FAKE_ID_1/Details"
          >
            Dinner for me and the gals, PIZZAA!
          </Text>
        </View>
        <View
          style={Object {}}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                  },
                  Object {
                    "backgroundColor": "#7AD6FE",
                    "borderRadius": 20,
                    "height": 40,
                    "width": 40,
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontFamily": "Inter-Medium",
                      "fontSize": 16,
                    },
                    Object {
                      "fontSize": 20,
                    },
                  ]
                }
              >
                0
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
        testID="OutgoingPaymentRequestNotification/FAKE_ID_1/CallToActions"
      >
        <View
          accessible={true}
          focusable={true}
          nativeBackgroundAndroid={
            Object {
              "attribute": "selectableItemBackgroundBorderless",
              "type": "ThemeAttrAndroid",
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="OutgoingPaymentRequestNotification/FAKE_ID_1/CallToActions/global:remind/Button"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#1AB775",
                  "fontFamily": "Inter-SemiBold",
                  "fontSize": 16,
                  "lineHeight": 22,
                },
                Object {
                  "fontSize": 14,
                  "lineHeight": 16,
                  "marginRight": 24,
                },
              ]
            }
          >
            global:remind
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          nativeBackgroundAndroid={
            Object {
              "attribute": "selectableItemBackgroundBorderless",
              "type": "ThemeAttrAndroid",
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="OutgoingPaymentRequestNotification/FAKE_ID_1/CallToActions/global:cancel/Button"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#1AB775",
                  "fontFamily": "Inter-SemiBold",
                  "fontSize": 16,
                  "lineHeight": 22,
                },
                Object {
                  "fontSize": 14,
                  "lineHeight": 16,
                  "marginRight": 24,
                },
              ]
            }
          >
            global:cancel
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`OutgoingPaymentRequestSummaryNotification renders correctly for just one alone 1`] = `
<View>
  <View
    style={
      Object {
        "marginBottom": 16,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "padding": 16,
          },
          Object {
            "borderRadius": 8,
          },
          Object {
            "elevation": 12,
            "shadowColor": "rgba(156, 164, 169, 0.4)",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 12,
          },
          Array [
            Object {
              "height": 144,
            },
            Object {},
          ],
        ]
      }
      testID="OutgoingPaymentRequestNotification/FAKE_ID_1"
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "marginRight": 16,
            }
          }
        >
          <Text
            numberOfLines={2}
            style={
              Object {
                "color": "#9CA4A9",
                "fontFamily": "Inter-Regular",
                "fontSize": 13,
                "lineHeight": 16,
                "marginBottom": 4,
              }
            }
            testID="OutgoingPaymentRequestNotification/FAKE_ID_1/Title"
          >
            outgoingPaymentRequestNotificationTitle
          </Text>
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "#2E3338",
                "fontFamily": "Inter-SemiBold",
                "fontSize": 24,
                "lineHeight": 28,
                "marginBottom": 2,
              }
            }
            testID="OutgoingPaymentRequestNotification/FAKE_ID_1/Amount"
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  undefined,
                  Object {
                    "color": undefined,
                  },
                ]
              }
            >
              
              $
              266,000.00
            </Text>
          </Text>
          <Text
            style={
              Object {
                "color": "#2E3338",
                "fontFamily": "Inter-Regular",
                "fontSize": 14,
                "lineHeight": 18,
              }
            }
            testID="OutgoingPaymentRequestNotification/FAKE_ID_1/Details"
          >
            Dinner for me and the gals, PIZZAA!
          </Text>
        </View>
        <View
          style={Object {}}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                  },
                  Object {
                    "backgroundColor": "#7AD6FE",
                    "borderRadius": 20,
                    "height": 40,
                    "width": 40,
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontFamily": "Inter-Medium",
                      "fontSize": 16,
                    },
                    Object {
                      "fontSize": 20,
                    },
                  ]
                }
              >
                +
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
        testID="OutgoingPaymentRequestNotification/FAKE_ID_1/CallToActions"
      >
        <View
          accessible={true}
          focusable={true}
          nativeBackgroundAndroid={
            Object {
              "attribute": "selectableItemBackgroundBorderless",
              "type": "ThemeAttrAndroid",
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="OutgoingPaymentRequestNotification/FAKE_ID_1/CallToActions/global:remind/Button"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#1AB775",
                  "fontFamily": "Inter-SemiBold",
                  "fontSize": 16,
                  "lineHeight": 22,
                },
                Object {
                  "fontSize": 14,
                  "lineHeight": 16,
                  "marginRight": 24,
                },
              ]
            }
          >
            global:remind
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          nativeBackgroundAndroid={
            Object {
              "attribute": "selectableItemBackgroundBorderless",
              "type": "ThemeAttrAndroid",
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="OutgoingPaymentRequestNotification/FAKE_ID_1/CallToActions/global:cancel/Button"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#1AB775",
                  "fontFamily": "Inter-SemiBold",
                  "fontSize": 16,
                  "lineHeight": 22,
                },
                Object {
                  "fontSize": 14,
                  "lineHeight": 16,
                  "marginRight": 24,
                },
              ]
            }
          >
            global:cancel
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`OutgoingPaymentRequestSummaryNotification renders correctly for multiple requests 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "padding": 16,
      },
      Object {
        "borderRadius": 8,
      },
      Object {
        "elevation": 12,
        "shadowColor": "rgba(156, 164, 169, 0.4)",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 12,
      },
      Array [
        Object {
          "height": 144,
        },
        Object {},
      ],
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "width": 40,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../packages/mobile/src/images/send-dollar.png",
          }
        }
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "space-between",
          "marginLeft": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#2E3338",
            "fontFamily": "Inter-Medium",
            "fontSize": 16,
            "lineHeight": 22,
            "marginBottom": 4,
          }
        }
        testID="undefined/Title"
      >
        outgoingPaymentRequestsSummaryTitle
      </Text>
      <Text
        numberOfLines={3}
        style={
          Object {
            "color": "#81868B",
            "flex": 1,
            "fontFamily": "Inter-Regular",
            "fontSize": 14,
            "lineHeight": 18,
          }
        }
        testID="undefined/Details"
      />
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
        testID="undefined/CallToActions"
      >
        <View
          accessible={true}
          focusable={true}
          nativeBackgroundAndroid={
            Object {
              "attribute": "selectableItemBackgroundBorderless",
              "type": "ThemeAttrAndroid",
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="undefined/CallToActions/viewAll/Button"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#1AB775",
                  "fontFamily": "Inter-SemiBold",
                  "fontSize": 16,
                  "lineHeight": 22,
                },
                Object {
                  "fontSize": 14,
                  "lineHeight": 16,
                  "marginRight": 24,
                },
              ]
            }
          >
            viewAll
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
